rust/psibase_macros/src/graphql_macro.rs

```diff
@@ -0,0 +1,164 @@
+use darling::ToTokens;
+use proc_macro::TokenStream;
+use proc_macro2::Ident;
+use proc_macro_error::abort;
+use quote::quote;
+use syn::{
+    parse::{Parse, ParseStream},
+    parse_macro_input,
+    punctuated::Punctuated,
+    AttrStyle, ImplItem, Item, Result, Token,
+};
+
+struct Args {
+    query_name: Ident,
+    table: Ident,
+    key_fn: Ident,
+    sub_index_params: Vec<(Ident, Ident)>,
```

Comment:
Should we allow types that are represented by more than one token, i.e. using syn::TypeParam?

Reply:
Great idea, done!